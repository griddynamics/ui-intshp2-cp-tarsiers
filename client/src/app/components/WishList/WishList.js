/* eslint-disable object-curly-newline */
import React from 'react';
import { PropTypes } from 'prop-types';
import Carousel from '../../shared/Carousel';
import ProductItemContainer from '../../shared/ProductItem/ProductItemContainer';
import { isAddedToCart } from '../../../utils/inCartInWishlist.service';

import styles from './WishList.module.scss';

const title = 'Wish list';
const message = 'Currently your wishlist is empty. Add products to it first';

const WishList = props => {
  const { wishlist, cart } = props;

  const titleArr = title.split(' ');

  const list =
    wishlist.length &&
    wishlist.map(el => (
      <ProductItemContainer
        key={el._id}
        data={el}
        isAddedToCart={isAddedToCart(el._id, cart)}
      />
    ));

  return !list.length ? null : (
    <section className={styles.wishlist}>
      <div className="section_heading">
        <h2>
          <span className="highlighted">{titleArr[0]}</span>
          &nbsp;
          {titleArr[1]}
        </h2>
        {!wishlist.length ? <p>{message}</p> : ''}
      </div>
      <div className={styles.wishlist__display}>
        <Carousel data={wishlist}>{list}</Carousel>
      </div>
    </section>
  );
};

WishList.propTypes = {
  wishlist: PropTypes.arrayOf(PropTypes.object)
};

WishList.defaultProps = {
  wishlist: []
};

export default WishList;
