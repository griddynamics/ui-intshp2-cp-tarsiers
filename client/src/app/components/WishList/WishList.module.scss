@import '../../../styles/variables';
@import '../../../styles/mixin';

.wishlist {
  padding-top: 60px;
  padding-bottom: 120px;
  color: $color-dark;
  text-align: center;
  margin: 0 auto;
  max-width: $media-large;
  background-color: $color-light-1;

  &__display {
    margin: 0 auto;
    text-align: center;
    @include d-flex() {
      align-items: center;
    }
  }

  .wishlist-title {
    padding-bottom: 20px;
    font-size: 40px;
    color: $color-main;
    font-family: $main-font-bold;
    text-transform: uppercase;
    span {
      color: $color-dark;
    }
  }
}

.wishlist-block {
  @include d-flex(space-around);
  justify-content: unset;
  transition: 0.7s;
  .item {
    margin: 0 15px;
    display: flex;
    min-width: 370px;
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
    background-color: $color-white;

    &:first-child {
      margin: 0 15px 0 0;
    }
    &:hover {
      box-shadow: 0 0 10px rgba(0, 0, 0, 0.3);
    }

    .item-pic img {
      padding: 36px;
      width: 150px;
      max-width: initial;
    }

    .item-info {
      margin: 0 auto;
      padding: 5px;
      @include d-flex(center, column);
    }

    &__display {
      margin: 0 auto;
      text-align: center;
      @include d-flex() {
        align-items: center;
      }
    }

    &__block {
      @include d-flex(space-around);
      justify-content: unset;
      transition: 0.7s;
    }
  }
}
