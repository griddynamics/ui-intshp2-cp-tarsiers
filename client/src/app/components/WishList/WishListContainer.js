import { connect } from 'react-redux';
import WishList from './WishList';

const mapStateToProps = state => ({
  products: state.products.products,
  wishlist: state.wishlist.wishlist,
  cart: state.cart
});

const WishlistContainer = connect(mapStateToProps)(WishList);

export default WishlistContainer;
