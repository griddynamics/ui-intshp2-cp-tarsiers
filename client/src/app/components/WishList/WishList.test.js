import React from 'react';
import { shallow } from 'enzyme';
import WishList from './WishList';

describe('<WishList />', () => {
  it('should match its snapshot', () => {
    const props = { products: [], extended: false };
    const wrapper = shallow(<WishList {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
