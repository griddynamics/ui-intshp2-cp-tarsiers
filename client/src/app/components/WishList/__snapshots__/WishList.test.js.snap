// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WishList /> should match its snapshot 1`] = `""`;
